processor/src/main/resources/org/mapstruct/ap/internal/model/MapKeyMappingConstructorFragment.ftl

```diff
@@ -0,0 +1,12 @@
+<#--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<#-- @ftlvariable name="" type="org.mapstruct.ap.internal.model.MapKeyMappingConstructorFragment" -->
+this.${field.variableName} = new java.util.HashMap<>();
```

Comment:
For this you can then use an `IterableCreation` in the `MapKeyMappingConstructorFragment`. Most likely we'll need a new method in `IterableCreation`

e.g.

```java
    public static IterableCreation create(Type resultType) {
        return new IterableCreation( resultType, null, null );
    }
```

I think that the new line will then be 
```freemarker
this.${field.variableName} = <@includeModel object=iterableCreation useSizeIfPossible=false/>;
```